pandapipes/topology/create_graph.py

```diff
@@ -22,38 +25,24 @@
 
 
 def create_nxgraph(net, include_pipes=True, include_valves=True, include_pumps=True,
-                   nogojunctions=None, notravjunctions=None, multi=True,
-                   include_out_of_service=False):
+                   nogojunctions=None, notravjunctions=None, multi=True, neglect_in_service=False):
 
     if multi:
         mg = nx.MultiGraph()
     else:
         mg = nx.Graph()
 
-    if hasattr(net, "pipe"):
-        pipe = get_edge_table(net, "pipe", include_pipes)
-        if pipe is not None:
-            indices, parameter, in_service = init_par(pipe)
-            indices[:, F_JUNCTION] = pipe.from_junction.values
-            indices[:, T_JUNCTION] = pipe.to_junction.values
-            parameter[:, WEIGHT] = pipe.length_km.values
-            add_edges(mg, indices, parameter, in_service, net, "pipe")
-
-    if hasattr(net, "valve"):
-        valve = get_edge_table(net, "valve", include_valves)
-        if valve is not None:
-            indices, parameter, in_service = init_par(valve)
-            indices[:, F_JUNCTION] = valve.from_junction.values
-            indices[:, T_JUNCTION] = valve.to_junction.values
-            add_edges(mg, indices, parameter, in_service, net, "valve")
-
-    if hasattr(net, "pump"):
-        pump = get_edge_table(net, "pump", include_pumps)
-        if pump is not None:
-            indices, parameter, in_service = init_par(pump)
-            indices[:, F_JUNCTION] = pump.from_junction.values
-            indices[:, T_JUNCTION] = pump.to_junction.values
-            add_edges(mg, indices, parameter, in_service, net, "pump")
+    if neglect_in_service is False:
+        neglect_in_service_comps = []
+    elif neglect_in_service is True:
+        neglect_in_service_comps = [comp.table_name() for comp in net.component_list]
+    else:
+        neglect_in_service_comps = list(neglect_in_service)
+
+    for comp, weight_col, include_comp in zip([Pipe, Valve, Pump], ["length_km", None, None],
+                                              [include_pipes, include_valves, include_pumps]):
```

Comment:
What if someone creates another, new branch_element and wants to add this to the nx_graph. This, as I can see it, is not possible at the moment. How about adding a **kwargs, where you can pass a list of additional components you want to add.

Reply:
Yes, I thought about that as well... but I haven't yet found a way to insert the weight behavior here. I am sure, we can come up with a nice solution, especially since we need to combine it with your previous comment.

Reply:
OK, here's a suggestion: We always add all branch components and respect the in_service flag. Only if clearly indicated otherwise in the kwargs, we alter the behavior:

    for comp in net.component_list:
        if not issubclass(comp, BranchComponent):
            continue
	tbl = comp.table_name()
	include_comp = kwargs.get("consider_%s" % tbl, True)
	respect_in_service = kwargs.get("respect_%s" % tbl, True)
	# some formulation to add weight
	weight_col = ??
	add_branch_component(comp, mg, net, include_comp, respect_in_service, weight_col)

And then we just need to rewrite the `add_branch_component` function.

Reply:
Maybe don't use it in kwargs, but rather do it like included_components=None and weight='length_km'. Alternatively, there you can specify included_components and weight indivudally. included_components can be a list then and the weight-variable must be a list of same length, too. If included_components is None, all branch_components are taken into account and 'legnth_km' is taken as default. What do you think or do you think it is too implicit?